layer/context.py

```diff
@@ -81,7 +90,9 @@ def asset_type(self) -> AssetType:
             raise Exception("Unsupported asset type")
 
     def close(self) -> None:
-        pass
+        log_destination = self.log_data_destination()
```

Comment:
Why do we go through the public method instead of directly accessing the instance var?

Reply:
this got changed but you're right, this didn't make sense. hopefully no longer a case anywhere :)